### PR TITLE
Fix resource loading

### PR DESCRIPTION
**Pull request by Chocohead**
Fixes the resource downloading thread by using the asset index location which Fabric passes in

### CODE REVIEW

**Review by valoeghese — APPROVED** [review]
but have you tested in prod

File: src/main/java/io/github/minecraftcursedlegacy/mixin/MixinResourceDownloadThread.java [hunk 1]
```diff
@@ -1,20 +1,106 @@
...
+					//The vanilla launcher will use this as the index name
+					assetList = index.resolve("pre-1.6.json");
+
+					if (!Files.isReadable(assetList)) {
+						//Loom however prefixes it with the Minecraft version too
+						assetList = index.resolve("b1.7.3-pre-1.6.json");
+					}
+
+					if (!Files.isReadable(assetList)) {
+						System.err.println("Unable to find asset index");
```
Comment (williambl) [inline src/main/java/io/github/minecraftcursedlegacy/mixin/MixinResourceDownloadThread.java:56]: Is there not a logger available?

**Review by williambl — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/minecraftcursedlegacy/mixin/MixinResourceDownloadThread.java [hunk 2]
```diff
@@ -1,20 +1,106 @@
...
+					}
+				} else {
+					System.err.println("Cannot find object directory (expected it to be " + files + ')');
+					break out;
+				}
+			} else {
+				System.err.println("Assets directory not supplied, skipping loading");
+			}
+
+			info.cancel();
```
Comment (CoolMineman) [inline src/main/java/io/github/minecraftcursedlegacy/mixin/MixinResourceDownloadThread.java:103]: Why not @overwrite if you inject head and return every time?

File: src/main/java/io/github/minecraftcursedlegacy/mixin/MixinResourceDownloadThread.java [hunk 3]
```diff
@@ -1,20 +1,106 @@
...
+	@Shadow
+	public File workingDirectory;
+	@Shadow
+	private Minecraft field_138;
+	@Shadow
+	private boolean field_139;
+
+	@Inject(method = "run", at = @At("HEAD"), remap = false, cancellable = true)
+	private void go(CallbackInfo info) {
+		if (field_138.isApplet && ((AccessorMinecraft) field_138).getApplet() != null) {
```
Comment (CoolMineman) [inline src/main/java/io/github/minecraftcursedlegacy/mixin/MixinResourceDownloadThread.java:38]: Why just applet? Could we support the normal launch too?

**Review by CoolMineman — CHANGES_REQUESTED** [review]
Logger, Non-applet, and cursed inject head and cancel every time.

**Review by CoolMineman — APPROVED** [review]
nvm good and best

**Review by williambl — APPROVED** [review]

**Pull request merged**